### PR TITLE
Added lm_split function and documentation

### PR DESCRIPTION
**Pull request by ghost**
Added lm_split() to the "utils" list of functions
-Includes documentation for this new function
Corrected "Check" warning
-Was missing parameters for war_frame() function

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -72,3 +72,35 @@ standardize_team_ids <- function(x) {
...
+#' @param equal_width If TRUE, cuts the data based on even widths of value; if FALSE, each cut will have an approximately equal number of observations
+#' @param ... Passes lm() parameters to allow user to handle NA values
+#' @examples lm_split(comps_hypercube, "rWAR ~ cum_rwar",i = 4, "cum_rwar", na.omit = TRUE)
+#' @export
+
+lm_split <- function(data, form, i, var_name, equal_width = TRUE, ...) {
+  if (equal_width == TRUE) {
+    #calculates splits for even variable bin width
+    data$splits <- cut(data[[var_name]], i, include.lowest = TRUE)
+  } else if (equal_width == FALSE){
```
Comment (beanumber) [inline R/utils.R]: I don't think you need the second `if()`, because if `equal_width` isn't `TRUE`, it is necessarily `FALSE`.

File: R/utils.R [hunk 2]
```diff
@@ -72,3 +72,35 @@ standardize_team_ids <- function(x) {
...
+  #splits data frame based on new split variable
+  splitframe <- split(data, data[["splits"]])
+
+  #runs the model on the list of split data frames
+  mapframe <- map(splitframe, lm, formula = form)
+
+  #creates model summaries for each data frame and adds a variable to show splits
+  glanceframe <- mapframe %>%
+    map_dfr(glance, .id = paste(var_name, "splits", sep = "_"))
+}
```
Comment (beanumber) [inline R/utils.R:107]: Is this the end of the function? Don't we want to return `glanceframe`?

File: R/war.R [hunk 3]
```diff
@@ -1,5 +1,7 @@
 #' Build a data frame suitable for projecting WAR
 #' @export
+#' @param year Predict until this year
+#' @param ... Can pass logicals for the filter function
```
Comment (beanumber) [inline R/war.R:4]: Thanks!

**Review by beanumber — CHANGES_REQUESTED** [review]
Can you please add yourself as a `ctb` in `DESCRIPTION`?

File: R/utils.R (hunk 1 above)
Comment (ghost) [reply R/utils.R]: Done!

File: R/utils.R (hunk 2 above)
Comment (ghost) [reply R/utils.R:107]: Got it; changed!

**Comment by beanumber** [comment]
Thanks @stannir98 !

**Pull request merged**